exopy_hqc_legacy/instruments/drivers/visa/agilent_pna.py

```diff
@@ -80,14 +80,26 @@ def read_formatted_data(self, meas_name=''):
             self.selected_measure = meas_name
         else:
             meas_name = self.selected_measure
+            
+        # get the endianess of the output so that the data is read properly
+        data_endianess = self._pna.data_endianess
+        if data_endianess == 'little':
+            is_big_endian = False
+        elif data_endianess == 'big':
+            is_big_endian = True
+        else:
+            raise InstrIOError(cleandoc('''Agilent PNA did not define 
+                                        endianess properly'''))
 
         data_request = 'CALCulate{}:DATA? FDATA'.format(self._channel)
-        if self._pna.data_format == 'REAL,32':
-            data = self._pna.query_binary_values(data_request, 'f')
-
+        if self._pna.data_format in ('REAL,32', 'REAL,+32'):
+            data = self._pna.query_binary_values(data_request, datatype='f', 
+                                                 is_big_endian=is_big_endian)
+            
         elif self._pna.data_format in ('REAL,64', 'REAL,+64'):
```

Comment:
Honestly you should make `data_format` return a better defined value since you have control over it to avoid dealing with that kind of random stuff all over the place.

Reply:
I agree, but the weird stuff is that in the doc http://na.support.keysight.com/pna/help/latest/Programming/GP-IB_Command_Finder/Format_SCPI.htm#fb I understand the different formats to be 'REAL,32', 'REAL,64', and 'ASCii,0' while they are in fact 'REAL,+32', 'REAL,+64', and 'ASC,+0'.
Shall I give up on the syntax defined in the doc?

Reply:
Ok in fact I see what you mean, let me include all this checks and name conversion in data format

Reply:
You can keep both but do the conversion inside `data_format` to avoid dealing with it in possibly multiple places afterwards.